### PR TITLE
cube: Simplify delete operations

### PR DESCRIPTION
**Pull request by stweil**
It is not necessary to check for null pointers.

Remove also unneeded delete operations and add missing delete operations
in cube/bmp_8.cpp.

Simplify also a conditional statement in cube/cube_object.cpp.

Signed-off-by: Stefan Weil <sw@weilnetz.de>

### CODE REVIEW

**Comment by stweil** [comment]
This fixes issue #97.

File: cube/bmp_8.cpp [hunk 1]
```diff
@@ -77,14 +73,14 @@ unsigned char **Bmp8::CreateBmpBuffer(unsigned char init_val) {
 
   buff = (unsigned char **) new unsigned char *[hgt_ * sizeof(*buff)];
   if (!buff) {
```
Comment (elfring) [inline cube/bmp_8.cpp]: I find this null pointer check after the use of the C++ new operator questionable.
Can it be omitted here?

**Comment by elfring** [comment]
Thanks for [your small source code improvement](https://github.com/tesseract-ocr/tesseract/pull/507/commits/c46e773758f1940d8bc4733ee6d119cc60d5caaa "Simplify delete operations").

File: cube/bmp_8.cpp (hunk 1 above)
Comment (stweil) [inline cube/bmp_8.cpp]: Yes, indeed. Do you want to send a pull request?

File: cube/bmp_8.cpp [hunk 2]
```diff
@@ -77,14 +73,14 @@ unsigned char **Bmp8::CreateBmpBuffer(unsigned char init_val) {
...
-    delete []buff;
     return NULL;
   }
 
   // alloc and init memory for buffer and line buffer
   buff[0] = (unsigned char *)
       new unsigned char[stride_ * hgt_ * sizeof(*buff[0])];
   if (!buff[0]) {
```
Comment (stweil) [inline cube/bmp_8.cpp]: @elfring, this test can also be removed completely.

File: cube/bmp_8.cpp [hunk 3]
```diff
@@ -105,13 +101,13 @@ unsigned int ** Bmp8::CreateBmpBuffer(int wid, int hgt,
   // compute stride (align on 4 byte boundries)
   buff = (unsigned int **) new unsigned int *[hgt * sizeof(*buff)];
   if (!buff) {
-    delete []buff;
     return NULL;
   }
 
   // alloc and init memory for buffer and line buffer
   buff[0] = (unsigned int *) new unsigned int[wid * hgt * sizeof(*buff[0])];
   if (!buff[0]) {
```
Comment (stweil) [inline cube/bmp_8.cpp]: Should be removed completely as well.

File: cube/bmp_8.cpp (hunk 1 above)
Comment (elfring) [inline cube/bmp_8.cpp]: * Do any special build options matter for the handling of C++ exceptions by such a software component?
* Can any smart pointers help to improve the source code further?

Comment (amitdo) [inline cube/bmp_8.cpp]: The codebase doesn't have exceptions handling (Goole have a C++ policy that forbid their use).

Comment (amitdo) [inline cube/bmp_8.cpp]: >Can any smart pointers help to improve the source code further?
There is a disscution about thos topic. See https://groups.google.com/forum/#!topic/tesseract-dev/mZ2IUsvWgbY

Comment (zdenop) [inline cube/bmp_8.cpp]: cube will be removed from source code... I would like to keep it for 3.05 at least. 
So only bugfixes for cube make sense for me..

**Pull request merged**